lib/writer.js

```diff
@@ -108,10 +108,10 @@
       return `${inh.trivia.colon}:${inh.trivia.name}${inh.name}`;
     }
 
-    function interface_(it) {
+    function interface_like(it, type = it.type) {
```

Comment:
I'm not a huge fan of taking the default value from the outer scope, tbh. I think we should just make the second argument mandatory. 

I wonder if we should actually call this "container" instead of "interface_like"... though it's true that they are "interface like", the thing that makes this the same is that they are the main container types.